src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -647,13 +663,20 @@ public void Docs_Backtracking_WithOptionalQuantifiersOrAlternationConstructs(Reg
 
         // https://docs.microsoft.com/en-us/dotnet/standard/base-types/backtracking-in-regular-expressions#nonbacktracking-subexpression
         [Theory]
-        [InlineData(RegexOptions.None)]
-        [InlineData(RegexOptions.Compiled)]
+        [MemberData(nameof(RegexOptions_TestData))]
         public void Docs_Backtracking_WithNestedOptionalQuantifiers(RegexOptions options)
         {
             const string Input = "b51:4:1DB:9EE1:5:27d60:f44:D4:cd:E:5:0A5:4a:D24:41Ad:";
             // Assert.False(Regex.IsMatch(Input, "^(([0-9a-fA-F]{1,4}:)*([0-9a-fA-F]{1,4}))*(::)$")); // takes too long due to backtracking
-            Assert.False(Regex.IsMatch(Input, "^((?>[0-9a-fA-F]{1,4}:)*(?>[0-9a-fA-F]{1,4}))*(::)$", options)); // non-backtracking
+            if (options == RegexHelpers.RegexOptionNonBacktracking)
+            {
+                // With NonBacktracking option this test becomes feasible: from almost a minute to less than a second
+                Assert.False(Regex.IsMatch(Input, "^(([0-9a-fA-F]{1,4}:)*([0-9a-fA-F]{1,4}))*(::)$", options));
+            }
+            else
+            {
+                Assert.False(Regex.IsMatch(Input, "^((?>[0-9a-fA-F]{1,4}:)*(?>[0-9a-fA-F]{1,4}))*(::)$", options)); // forced non-backtracking
+            }
```

Comment:
```suggestion
            Assert.False(Regex.IsMatch(Input, "^((?>[0-9a-fA-F]{1,4}:)*(?>[0-9a-fA-F]{1,4}))*(::)$", options)); // forced non-backtracking
```

Reply:
The atomic loop annotation  `(?>...)` is not supported in `NonBacktracking` mode. So the `IsMatch` will throw an exception if not guarded by `else`.

Reply:
Ah, ok.  Can you remind me why?  Rather than writing it one-off here, I think we should add a README.md to the root of the SRM folder that outlines:
- What features aren't supported (and why)?
- Key terminology employed in the implementation
- Other interesting details to help understand the implementation

We can do that subsequently.

Reply:
Short reply is that it changes the meaning of the loop -- cutting of choices that would exist with backtracking. In `NonBacktracking` mode it messes with/contradicts the DFA semantics because all search paths are preserved there -- just that backtracking is not being used to preserve those paths. We discussed this briefly at some point in the spring. Yes, **documentation** is the next big item on the todo list.